dart/lib/src/open_location_code.dart

```diff
@@ -118,13 +120,9 @@ bool isValid(String code) {
     return false;
   }
 
-  // Check code contains only valid characters.
-  for (var ch in code.codeUnits) {
-    if (ch > _decode.length || _decode[ch] < -1) {
-      return false;
-    }
-  }
-  return true;
+  var filterCallback = (ch) => ch > _decode.length || _decode[ch] < -1;
+
+  return null == code.codeUnits.firstWhere(filterCallback, orElse: () => null);
```

Comment:
Instead of using a combination of firstWhere and checking null from the orElse, use the every method instead, which simply returns a boolean.

https://api.dartlang.org/stable/2.1.0/dart-core/Iterable/every.html